cms/djangoapps/contentstore/tasks.py

```diff
@@ -125,3 +152,219 @@ def push_course_update_task(course_key_string, course_subscription_id, course_di
+    try:
+        tar_file = tarfile.open(temp_filepath)
+        try:
+            safetar_extractall(tar_file, (course_dir + u'/').encode(u'utf-8'))
+        except SuspiciousOperation as exc:
+            LOGGER.info(u'Course import %s: Unsafe tar file - %s', courselike_key, exc.args[0])
+            with respect_language(language):
+                self.status.fail(_(u'Unsafe tar file. Aborting import.'))
+            return
+        finally:
+            tar_file.close()
+
+        LOGGER.info(u'Course import %s: Uploaded file extracted', courselike_key)
+        self.status.set_state(u'Verifying')
+        self.status.increment_completed_steps()
+
+        # find the 'course.xml' file
+        def get_all_files(directory):
+            """
+            For each file in the directory, yield a 2-tuple of (file-name,
+            directory-path)
+            """
+            for directory_path, _dirnames, filenames in os.walk(directory):
+                for filename in filenames:
+                    yield (filename, directory_path)
+
+        def get_dir_for_filename(directory, filename):
+            """
+            Returns the directory path for the first file found in the directory
+            with the given name.  If there is no file in the directory with
+            the specified name, return None.
+            """
+            for name, directory_path in get_all_files(directory):
+                if name == filename:
+                    return directory_path
+            return None
+
+        dirpath = get_dir_for_filename(course_dir, root_name)
+        if not dirpath:
+            with respect_language(language):
+                self.status.fail(_(u'Could not find the {0} file in the package.').format(root_name))
+                return
+
+        dirpath = os.path.relpath(dirpath, data_root)
+        LOGGER.debug(u'found %s at %s', root_name, dirpath)
+
+        LOGGER.info(u'Course import %s: Extracted file verified', courselike_key)
+        self.status.set_state(u'Updating')
+        self.status.increment_completed_steps()
+
+        with dog_stats_api.timer(
+            u'courselike_import.time',
+            tags=[u"courselike:{}".format(courselike_key)]
+        ):
+            courselike_items = import_func(
+                modulestore(), user.id,
+                settings.GITHUB_REPO_ROOT, [dirpath],
+                load_error_modules=False,
+                static_content_store=contentstore(),
+                target_id=courselike_key
+            )
+
+        new_location = courselike_items[0].location
+        LOGGER.debug(u'new course at %s', new_location)
+
+        LOGGER.info(u'Course import %s: Course import successful', courselike_key)
+        self.status.increment_completed_steps()
+    except Exception as exception:   # pylint: disable=broad-except
+        LOGGER.exception(u'error importing course')
+        self.status.fail(text_type(exception))
```

Comment:
Here's another case of, I think, putting raw exceptions where users can see them. I'd log the exception with whatever relevant information might be needed to find it (`courselike_key` / user?) and tell the user to retry or contact support since they probably can't do anything with the info anyway.